lib/server/ServerResponse.js

```diff
@@ -35,6 +53,17 @@ var ServerResponse = Response.extend({
     }
 });
 
+function addAllFields(passedHeaders, headers) {
+    var fields = Object.keys(passedHeaders);
+    var currentField;
+
+    while(fields.length) {
+        currentField = fields.shift();
+
+        headers[currentField] = passedHeaders[currentField];
+    }
```

Comment:
i coded it this way due to technical boredom. in practice, this code path would be called for an object with probably ~3 keys in any route so most any loop would be fine.

good point about pop over shift since the order doesn't matter. 

the fastest thing to do would be Object.keys and then a regular for or while loop according to this jsperf (https://jsperf.com/object-keys-iteration/38). do you know forEach to be especially faster than for/while loop?

i'll switch to a for loop